src/ResourceManager/Sql/Commands.Sql/Security/Cmdlet/Auditing/SetAzureSqlDatabaseAuditingPolicy.cs

```diff
@@ -0,0 +1,121 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Sql.Security.Model;
+using Microsoft.Azure.Commands.Sql.Security.Services;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Sql.Security.Cmdlet.Auditing
+{
+    /// <summary>
+    /// Sets the auditing policy properties for a specific database.
+    /// </summary>
+    [Cmdlet(VerbsCommon.Set, "AzureSqlDatabaseAuditingPolicy"), OutputType(typeof(DatabaseAuditingPolicyModel))]
+    public class SetAzureSqlDatabaseAuditingPolicy : SqlDatabaseAuditingCmdletBase
+    {
+
+        [Parameter(Mandatory = false)]
+        public SwitchParameter PassThru { get; set; }
+
+        /// <summary>
+        /// Gets or sets the names of the event types to use.
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "Event types to audit")]
+        [ValidateSet(Constants.DataAccess, Constants.SchemaChanges, Constants.DataChanges, Constants.SecurityExceptions, Constants.RevokePermissions, Constants.All, Constants.None, IgnoreCase = false)]
+        public string[] EventType { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the storage account to use.
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The name of the storage account")]
+        [ValidateNotNullOrEmpty]
+        public string StorageAccountName { get; set; }
```

Comment:
The Azure PowerShell users expect to pass in an `StorageAzureContext` which is standard along with what Azure PowerShell does. From customer perspective if your cmdlet does not take context then they are limited to use:
- specific endpoint
- no support for authentication other than key
- They have to pass the hard-coded storage account name everywhere in the scripts
- The parameter name `StorageAccountName` leads to the problem parameters inconsistency as some people will refer it as `StorageAccount`

Please use `StorageContext` if there's no technical limitations.